packages/components/src/unit-control/index.native.js

```diff
@@ -20,125 +20,133 @@ import { CSS_UNITS } from './utils';
 /**
  * WordPress dependencies
  */
-import { useRef } from '@wordpress/element';
+import { useRef, useCallback, useMemo, memo } from '@wordpress/element';
 import { withPreferredColorScheme } from '@wordpress/compose';
 import { __, sprintf } from '@wordpress/i18n';
 
-function UnitControl( {
-	currentInput,
-	label,
-	value,
-	onChange,
-	onUnitChange,
-	initialPosition,
-	min,
-	max,
-	separatorType,
-	units = CSS_UNITS,
-	unit,
-	getStylesFromColorScheme,
-	decimalNum,
-	...props
-} ) {
-	const pickerRef = useRef();
-	const anchorNodeRef = useRef();
+const UnitControl = memo(
+	( {
+		currentInput,
+		label,
+		value,
+		onChange,
+		onUnitChange,
+		initialPosition,
+		min,
+		max,
+		separatorType,
+		units = CSS_UNITS,
+		unit,
+		getStylesFromColorScheme,
+		decimalNum,
+		...props
+	} ) => {
+		const pickerRef = useRef();
+		const anchorNodeRef = useRef();
 
-	function onPickerPresent() {
-		if ( pickerRef?.current ) {
-			pickerRef.current.presentPicker();
+		function onPickerPresent() {
+			if ( pickerRef?.current ) {
+				pickerRef.current.presentPicker();
+			}
 		}
-	}
 
-	const currentInputValue = currentInput === null ? value : currentInput;
-	const initialControlValue = isFinite( currentInputValue )
-		? currentInputValue
-		: initialPosition;
+		const currentInputValue = currentInput === null ? value : currentInput;
+		const initialControlValue = isFinite( currentInputValue )
+			? currentInputValue
+			: initialPosition;
+
+		const unitButtonTextStyle = getStylesFromColorScheme(
+			styles.unitButtonText,
+			styles.unitButtonTextDark
+		);
 
-	const unitButtonTextStyle = getStylesFromColorScheme(
-		styles.unitButtonText,
-		styles.unitButtonTextDark
-	);
+		/* translators: accessibility text. Inform about current unit value. %s: Current unit value. */
+		const accessibilityLabel = sprintf( __( 'Current unit is %s' ), unit );
 
-	const renderUnitButton = () => {
 		const accessibilityHint =
 			Platform.OS === 'ios'
 				? __( 'Double tap to open Action Sheet with available options' )
 				: __(
 						'Double tap to open Bottom Sheet with available options'
 				  );
 
-		/* translators: accessibility text. Inform about current unit value. %s: Current unit value. */
-		const accessibilityLabel = sprintf( __( 'Current unit is %s' ), unit );
+		const renderUnitButton = useMemo( () => {
```

Comment:
This `useMemo` could be removed if we have the `onPickerPresent` in dependencies. The `onPickerPresent` is created each time that the component is created/updated so the `renderUnitButton` also will be created. It would make more sense if the `useCallback` will be used.